core/cap-0046-02.md

```diff
@@ -199,29 +157,34 @@ functionality to support them.
 The following host functions are provided to instantiate contracts:
 
 ```rust
-// Instantiates a contract with the source referring to the provided wasm_hash.
-fn create_contract_from_contract(wasm_hash: Object /* 32-bytes array */,
-                                 salt: Object /* 32-bytes array */) -> Object /* 32-bytes array */
-// Instantiates a contract with the source referring to the built-in token.
-fn create_token_from_contract(salt: Object /* 32-bytes array */) -> Object /* 32-bytes array */
+// Uploads the WASM. Returns the SHA-256 hash of the WASM code.
+fn upload_wasm(wasm: Bytes) -> Bytes
+
+// Creates a WASM instance using the deployer, SHA-256 hash of the WASM, and a user specified salt.
+// Returns the Address of the newly created contract.
+fn create_contract(deployer: Address, wasm_hash: Bytes, salt:Bytes) -> Address
+
+// Creates a Stellar Asset Contract for the XDR serialized asset passed in. Returns the Address
+// for the newly created contract.
+fn create_asset_contract(serialized_asset: Bytes) -> Address
 ```
 
-All of these functions return the identifier of the newly created contract.
+The contractIDs for the contracts created with `create_contract` and
+`create_asset_contract` are derived from `CONTRACT_ID_PREIMAGE_FROM_ADDRESS` and
+`CONTRACT_ID_PREIMAGE_FROM_ASSET` respectively.
```

Comment:
Please also add the authorization note - `create_contract` has to be authorized by `deployer` using the same auth payload as for the direct host function above.

Reply:
Done